controllers/component_controller.go

```
@@ -470,46 +480,51 @@ func (r *ComponentReconciler) generateGitops(ctx context.Context, req ctrl.Reque
+	if localGitopsGen {
+		err = gitopsjoblib.GenerateGitopsBase(context.Background(), r.Client, *component, ioutils.NewFilesystem(), generate.GitOpsGenParams{
+			Generator: r.Generator,
+			Token:     r.GitToken,
+			RemoteURL: gitOpsURL,
+			Branch:    gitOpsBranch,
+			Context:   gitOpsContext,
+		})
+		if err != nil {
+			return err
+		}
+	} else {
+		gitopsJobConfig := gitopsjob.GitOpsJobConfig{
+			OperationType: gitopsjob.GenerateBase,
+			RepositoryURL: component.Status.GitOps.RepositoryURL,
+			ResourceName:  component.GetName(),
+			Branch:        gitOpsBranch,
+			Context:       gitOpsContext,
+		}
 
-	kubernetesResources, err := devfile.GetResourceFromDevfile(log, compDevfileData, deployAssociatedComponents, component.Name, component.Spec.ContainerImage)
-	if err != nil {
-		log.Error(err, "unable to get kubernetes resources from the devfile outerloop components")
-		return err
-	}
+		// Create a Kubernetes job (with random string appended to name to ensure uniqueness) to generate the GitOps resources
+		jobName := component.GetName()
+		if len(jobName) > 57 {
+			jobName = component.GetName()[0:56]
+		}
+		jobName = jobName + util.GetRandomString(5, true)
+		jobNamespace := r.GitOpsJobNamespace
+		if jobNamespace == "" {
+			jobNamespace = component.Namespace
+		}
+		err = gitopsjob.CreateGitOpsJob(ctx, r.Client, r.GitToken, jobName, jobNamespace, component.Namespace, gitopsJobConfig)
+		if err != nil {
+			return err
+		}
 
-	// Generate and push the gitops resources
-	gitopsConfig := prepare.PrepareGitopsConfig(ctx, r.Client, *component)
-	mappedGitOpsComponent := util.GetMappedGitOpsComponent(*component, kubernetesResources)
+		// Wait for the job to succeed, error out if the 5 min timeout is reached
+		err = gitopsjob.WaitForJob(log, context.Background(), r.Client, r.GitOpsJobClientSet, jobName, jobNamespace, 5*time.Minute)
+		if err != nil {
+			return r.CleanUpJobAndReturn(log, jobName, jobNamespace, err)
```

if we return like `return r.CleanUpJobAndReturn()` then the `gitopsjob.WaitForJob()` err will be lost